zkevm-circuits/src/evm_circuit/execution/error_oog_sload_sstore.rs

```diff
@@ -0,0 +1,585 @@
+                    + cal_sload_gas_cost_for_assignment(true);
+            }
+
+            Self {
+                bytecode,
+                gas_cost,
+                ..Default::default()
+            }
+        }
+
+        pub fn new_for_sstore(
+            key: U256,
+            value: U256,
+            value_prev: U256,
+            original_value: U256,
+            is_warm: bool,
+        ) -> Self {
+            let mut bytecode = bytecode! {
+                PUSH32(value_prev)
+                PUSH32(key)
+                SSTORE
+            };
+            let sstore_gas_cost = cal_sstore_gas_cost_for_assignment(
+                value_prev,
+                original_value,
+                original_value,
+                false,
+            );
+            let mut gas_cost = 2 * OpcodeId::PUSH32.constant_gas_cost().0
+                + max(
+                    sstore_gas_cost,
+                    GasCost::SSTORE_SENTRY.0.checked_add(1).unwrap(),
+                );
+            if is_warm {
+                bytecode.append(&bytecode! {
+                    PUSH32(value)
+                    PUSH32(key)
+                    SSTORE
+                });
+                let sstore_gas_cost = cal_sstore_gas_cost_for_assignment(
+                    value_prev,
+                    original_value,
+                    original_value,
+                    true,
+                );
+                gas_cost += 2 * OpcodeId::PUSH32.constant_gas_cost().0
+                    + max(
+                        sstore_gas_cost,
+                        GasCost::SSTORE_SENTRY.0.checked_add(1).unwrap(),
+                    );
+            }
+
+            Self {
+                key,
+                value,
+                value_prev,
+                original_value,
+                is_warm,
+                gas_cost,
+                bytecode,
+            }
+        }
+    }
+
+    fn test_root(testing_data: &TestingData) {
+        let ctx = TestContext::<2, 1>::new(
+            None,
+            |accs| {
+                accs[0]
+                    .address(MOCK_ACCOUNTS[0])
+                    .balance(eth(10))
+                    .code(testing_data.bytecode.clone())
+                    .storage([(testing_data.key, testing_data.original_value)].into_iter());
+                accs[1].address(MOCK_ACCOUNTS[1]).balance(eth(10));
+            },
+            |mut txs, accs| {
+                // Decrease expected gas cost (by 1) to trigger out of gas error.
+                txs[0]
+                    .from(accs[1].address)
+                    .to(accs[0].address)
+                    .gas((GasCost::TX.0 + testing_data.gas_cost - 1).into());
+            },
+            |block, _tx| block.number(0xcafe_u64),
+        )
+        .unwrap();
+
+        CircuitTestBuilder::new_from_test_ctx(ctx).run();
+    }
+
+    fn test_internal(call_data_offset: usize, call_data_length: usize, testing_data: &TestingData) {
```

Comment:
seems `call_data_offset` and `call_data_length` don't affect test target,  suggest to remove .

Reply:
Deleted parameter `call_data_offset` and `call_data_length`, and hardcoded as `0x20` and `0x00` in `test_internal` function in [9245a20](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1147/commits/9245a20afd85c8d8d64c50e911fc76d6eacb8278).